backend/open_webui/utils/middleware.py

```diff
@@ -1815,6 +1816,8 @@ async def stream_body_handler(response, form_data):
                     nonlocal content_blocks
 
                     response_tool_calls = []
+                    delta_count = 0
+                    delta_batch_size = max(1, int(form_data.get("metadata", {}).get("stream_batch_size") or 1))
```

Comment:
```suggestion
                    delta_batch_size = max(MIN_STREAMING_BATCH_SIZE, int(form_data.get("metadata", {}).get("stream_batch_size") or 1))
```
And adding `MIN_STREAMING_BATCH_SIZE` as env (or similar).